lib/sim.c

```diff
@@ -907,12 +923,39 @@ sim_generate_parents(sim_t *self)
             goto out;
         }
     }
+    /* In case of a recombination event in the Kingman phase
+     * we require exactly one recombination point.
+     */
+    if(arg_flag == 1) {
+        node = C[0]->ancestry.head;
+        locus_mapping = (int_map_value_t *) node->item;
+        site_2 = (double)(locus_mapping->key);
+        node = node->next;
+        while(node != NULL) {
+            site_1 = site_2;
+            locus_mapping = (int_map_value_t *) node->item;
+            site_2 = (double)(locus_mapping->key);
+            recomb_rates[i] = site_2 - site_1;
+            total_rate += recomb_rates[i];
+            i++;
+            node = node->next;
+        } 
+        recomb_rates[0] /= total_rate;
+        for(j = 1; j < gap_no; j++) {
+            recomb_rates[j] = recomb_rates[j - 1] + recomb_rates[j] / total_rate;
+        }
+        double coin = gsl_rng_uniform(self->rng);
```

Comment:
In the rest of the code, variable declarations are at the top of each function. It would be nice to be consistent, but it's not a big deal.